var/spack/repos/builtin/packages/hip/package.py

```diff
@@ -498,5 +508,7 @@ def cmake_args(self):
             args.append(self.define("HIP_CATCH_TEST", "OFF"))
             args.append(self.define("ROCCLR_PATH", self.stage.source_path + "/rocclr"))
             args.append(self.define("AMD_OPENCL_PATH", self.stage.source_path + "/opencl"))
+        if "@5.3.0:" in self.spec:
+            args.append("-DCMAKE_INSTALL_LIBDIR=lib")
```

Comment:
`args.append("-DCMAKE_INSTALL_LIBDIR=lib")` should probably also be added unconditionally for rocsolver, hipsolver, rocBLAS, hipblas, rocfft, hipfft, rocsparse, hipsparse, and rocalution. Prior to ROCm 5.3, those libraries all used `set(CMAKE_INSTALL_LIBDIR "lib" CACHE INTERNAL "Installation directory for libraries" FORCE)` internally. That forceful assignment was removed in ROCm 5.3.

`rocm-cmake@5.3.0:` does set the default value to `lib` (without forcing it), but IMO package maintainers should be setting the value explicitly rather than depending on rocm-cmake (especially since older versions of rocm-cmake won't set it, but are likely to be otherwise compatible with the rocm mathlibs).

Reply:
Updated with -DCMAKE_INSTALL_LIBDIR=lib for rocsolver, rocblas, hipblas, rocfft, rocsparse, hipsparse and rocalution.
hipsolver and hipfft need to be updated under the #33405